### PR TITLE
Form validation

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**
Validates the inputs of users.
If the validation-check fails, the user has no chance to save his changes.

Validated forms:
- Topics
- Motions
- Users
- Workflows
- Categories
- MotionBlocks
- Tags
- CommentSection
- Elections
- MediaFiles

### CODE REVIEW

File: client/src/app/shared/components/head-bar/head-bar.component.ts [hunk 1]
```diff
@@ -71,6 +71,12 @@ export class HeadBarComponent {
     @Input()
     public editMode = false;
 
+    /**
+     * The save button can manually be hidden.
+     */
+    @Input()
+    public displaySaveButton = true;
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/head-bar/head-bar.component.ts]: Hmm, hidden and disabled i something different. Please be clear with the name of this input.

File: client/src/app/shared/validators/custom-validators.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { ValidatorFn, FormGroup, ValidationErrors } from '@angular/forms';
+
+export const durationValidator: ValidatorFn = (control: FormGroup): ValidationErrors | null => {
+    const regExp = /^\s*([0-9]+)(:)?([0-5][0-9]?)?\s*[h|m]?$/g;
```
Comment (FinnStutzenstein) [inline client/src/app/shared/validators/custom-validators.ts]: Comments... Pleae describe what this regex does (e.g. take it into parts and describe them, or do some examples)

File: client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts [hunk 3]
```diff
@@ -196,10 +196,10 @@ export class AssignmentDetailComponent extends BaseViewComponent implements OnIn
...
             tags_id: [],
             attachments_id: [],
-            title: '',
-            description: '',
-            poll_description_default: '',
-            open_posts: 0,
+            title: ['', Validators.required],
+            description: [''],
+            poll_description_default: [''],
+            open_posts: [0, Validators.required],
```
Comment (tsiegleauq) [inline client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts]: `open_posts` seems to have no effect. The form accepts this field, even tho it's empty.

File: client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html [hunk 4]
```diff
@@ -292,8 +293,11 @@ <h4>{{ 'Description' | translate }}:</h4>
                         matInput
                         placeholder="{{ 'Number of persons to be elected' | translate }}"
                         formControlName="open_posts"
+                        type="number"
+                        required
                         [value]="assignmentCopy.assignment.open_posts || null"
                     />
+                    <mat-error>{{ 'This field is required' | translate }}</mat-error>
```
Comment (tsiegleauq) [inline client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html]: Does currently not appear

File: client/src/app/site/agenda/components/item-info-dialog/item-info-dialog.component.html [hunk 5]
```diff
@@ -13,6 +13,7 @@ <h1 mat-dialog-title>{{ 'Edit details for' | translate }} {{ item.getTitle() }}<
         <!-- Duration -->
         <mat-form-field>
             <input type="string" matInput placeholder="{{ 'Duration' | translate }}" formControlName="durationText" />
+            <mat-hint *ngIf="agendaInfoForm.invalid">Your input does not match the following structure 'hh:mm'...</mat-hint>
```
Comment (tsiegleauq) [inline client/src/app/site/agenda/components/item-info-dialog/item-info-dialog.component.html]:
```suggestion
            <mat-error *ngIf="agendaInfoForm.invalid">Your input does not match the following structure 'hh:mm'...</mat-error>
```

**Pull request merged**